pkg/servicescm/servicescm_test.go

```diff
@@ -156,46 +156,61 @@ func TestGenerate(t *testing.T) {
 		name           string
 		services       []Service
 		files          []FileInfo
-		envVars        map[string]string
+		envVars0       map[string]string
+		envVars1       map[string]string
 		watchedEnvVars []string
+		expectedErr    bool
 	}{
 		{
-			name:     "non-empty envVars",
+			name:     "ConfigMaps with different envVar values",
 			services: testServices,
 			files:    testFiles,
-			envVars: map[string]string{
-				"HTTP_PROXY": "http://dev:ad2bc205af349589cb2c425daacf7a00@10.0.29.194:3128/",
-				"HTTS_PROXY": "http://dev:ad2bc205af349589cb2c425daacf7a00@10.0.29.194:3128/",
-				"NO_PROXY":   "localhost",
+			envVars0: map[string]string{
+				"HTTP_PROXY":  "proxy_endpoint:3128/",
+				"HTTPS_PROXY": "proxy_endpoint:3128/",
+				"NO_PROXY":    "localhost",
+			},
+			envVars1: map[string]string{
+				"HTTP_PROXY":  "proxy_endpoint:3128/",
+				"HTTPS_PROXY": "proxy_endpoint:3128/",
+				"NO_PROXY":    "localhost, example.com",
 			},
 			watchedEnvVars: testEnvVars,
+			expectedErr:    true,
 		},
 		{
 			name:           "empty envVars",
 			services:       testServices,
 			files:          testFiles,
-			envVars:        map[string]string{},
+			envVars0:       map[string]string{},
+			envVars1:       map[string]string{},
 			watchedEnvVars: testEnvVars,
+			expectedErr:    false,
 		},
 		{
 			name:           "nil envVars",
 			services:       testServices,
 			files:          testFiles,
 			watchedEnvVars: testEnvVars,
+			expectedErr:    false,
 		},
 	}
 
 	for _, test := range testCases {
 		t.Run(test.name, func(t *testing.T) {
-			data, err := NewData(&test.services, &test.files, test.envVars, test.watchedEnvVars)
+			data0, err := NewData(&test.services, &test.files, test.envVars0, test.watchedEnvVars)
 			require.NoError(t, err)
-			configMap, err := Generate(Name, "testNamespace", data)
+			data1, err := NewData(&test.services, &test.files, test.envVars1, test.watchedEnvVars)
+			configMap, err := Generate(Name, "testNamespace", data1)
 			require.NoError(t, err)
-
 			// Ensure that the ConfigMap we generate passes our own validation functions
 			parsed, err := Parse(configMap.Data)
 			require.NoError(t, err)
-			assert.NoError(t, parsed.ValidateExpectedContent(data))
+			if test.expectedErr {
+				assert.Error(t, parsed.ValidateExpectedContent(data0))
+				return
+			}
+			assert.NoError(t, parsed.ValidateExpectedContent(data0))
```

Comment:
I'm having a hard time following these test steps, and what specifically you are trying to test with these changes. It seems like you're adding a somewhat different test workflow in order to test a specific case, and trying to make it work for all the tests.


It might be better to test case as a subtest separate from the rest of the test cases.
```
 	for _, test := range testCases {
		t.Run(test.name, func(t *testing.T) { ... })
	}
	t.Run("new test being added in this PR) {
	}
 ```
 
Please also make the variable names more descriptive, envVars0/1 and data0/1 doesn't give any information about the difference between the two.